packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,131 @@
+	 * The only schema evolution that's currently possible is upgrading the schema to one that supports a superset of what the old schema allowed,
+	 * and collaborating between clients which have view schema that exactly correspond to that stored schema.
+	 * Future work on tree as well as these utilities should address this limitation.
+	 */
+	public abstract readonly config: TreeViewConfiguration<TSchema>;
+
+	/**
+	 * The underlying tree.
+	 */
+	#tree: ITree | undefined;
+
+	/**
+	 * View of the underlying tree.
+	 */
+	#treeView: TreeView<TSchema> | undefined;
+
+	/**
+	 * Gets the underlying {@link @fluidframework/tree#ITree}.
+	 * @throws If the tree has not yet been initialized, this will throw an error.
+	 */
+	private get initializedTree(): ITree {
+		if (!this.#tree) {
+			// Note: Can't use `UsageError` because adding dependency on `telemetry-utils` would create a cycle.
+			// TODO: would probably be useful to move our shared error types in a more accessible location.
+			throw new Error(this.getUninitializedErrorString("root"));
+		}
+
+		return this.#tree;
+	}
+
+	/**
+	 * Gets the view of the underlying {@link @fluidframework/tree#ITree}.
+	 * @throws If the tree has not yet been initialized, this will throw an error.
+	 */
+	public get tree(): TreeView<TSchema> {
+		if (this.#treeView === undefined) {
+			throw new Error(this.getUninitializedErrorString("tree"));
+		}
+		return this.#treeView;
+	}
+
+	public override async initializeInternal(existing: boolean): Promise<void> {
+		if (existing) {
+			// data store has a root tree so we just need to set it before calling initializingFromExisting
+			const channel = await this.runtime.getChannel(treeChannelId);
+			if (SharedTree.is(channel)) {
+				this.#tree = channel;
+			} else {
+				throw new Error(
+					`Content with id ${channel.id} is not a SharedTree and cannot be loaded with treeDataObject.`,
+				);
+			}
+		} else {
+			this.#tree = SharedTree.create(this.runtime, treeChannelId);
+			(this.#tree as unknown as ISharedObject).bindToContext();
+		}
+
+		await super.initializeInternal(existing);
+	}
+
+	protected override async initializingFirstTime(): Promise<void> {
+		this.#treeView = this.initializedTree.viewWith(this.config);
+
+		// Initialize the tree content and schema.
+		this.#treeView.initialize(this.createInitialTree());
+	}
+
+	protected override async initializingFromExisting(): Promise<void> {
+		this.#treeView = this.initializedTree.viewWith(this.config);
+	}
+
+	protected override async hasInitialized(): Promise<void> {
+		if (this.#treeView === undefined) {
+			throw new Error(this.getUninitializedErrorString("tree"));
+		}
+	}
+
+	/**
+	 * Generates an error string indicating an item is uninitialized.
+	 * @param item - The name of the item that was uninitialized.
+	 * @virtual
+	 */
+	protected getUninitializedErrorString(item: string): string {
```

Comment:
can this just be a free function, rather than existing on the class. everything on the class uses memory per-instance, and makes the api more complicates for developers to use

Reply:
Thats not how JavaScript works. This is a member on the prototype and thus a single copy of it exists. It gets passed the instance when invoked as a method as the `this` parameter and is found on the instance by searching the prototype chain.

Reply:
yeah. you are right. it still pollutes the class with an unnecessary member which i think makes the user experience worse

Reply:
Either way, I think I agree with the goal of simplifying the initial API here. I'll go ahead and remove this degree of freedom from the API for now - we can always re-add it later as needed.